lib/cache.js

```diff
@@ -22,12 +22,16 @@ class Cache {
 
   getPage(page) {
     let preferredPlatform = platform.getPreferredPlatformFolder(this.config);
-    return index.findPlatform(page, preferredPlatform)
+    let preferredLanguage = 'en';
+    if (process.env['LANG'] !== '') {
+      preferredLanguage = process.env['LANG'];
```

Comment:
Hrm, this is probably not what you would want. Our client specification does not follow [the POSIX locale specification](https://www.gnu.org/software/gettext/manual/html_node/Locale-Names.html#Locale-Names) for language environment variables. As a result of this, the `LANG` environment var is basically never going to be right, and should be converted first.

Some examples in case to make it clear:

- `LANG="it_IT.UTF-8"` and `LANG="it_IT"` should be converted to `it-IT`, then check for `pages.it-IT/platform/page.md` or `pages.it/platform/page.md`.
- `LANG="en_US.UTF-8"` and `LANG="en_US"` should be converted to `en-US`, then (special case since language code is `en`) check for `pages/platform/page.md`.
- `LANG="en.UTF-8"` and and `LANG="en"` should be converted to `en`, then (special case since language code is `en`) check for `pages/platform/page.md`.
- `LANG="sr_RS@latin"` should be converted to `sr-RS`, then check for `pages.sr-RS/platform/page.md` or `pages.sr/platform/page.md`.

---

This is an issue I've been thinking about for quite a while. I really think we should update the client spec and follow the POSIX locale specification (basically just use `ll_CC` instead of `ll-CC`). It would make it really easier to deal with languages.

Anyway, the fallback to `ll` from `ll_CC` (or `ll-CC`) if `pages.ll_CC` (or `pages.ll-CC`) is not present should be made regardless.

Reply:
I was thinking this earlier too. I know there was a reason that we chose the current language code format though, can't remember why.

Reply:
@pxgamer well, the only issue here is the dash `-` versus the underscore `_`. I don't remember a real reason for this choice, if I recall right :thinking:.

Reply:
It's here - https://gitter.im/tldr-pages/tldr?at=5c387be59289cc7aa757c1ad. You commented yourself @mebeim :) 

tldr is  GNU uses _ and BCP-47 uses -. They have different standards. And we decided to go with BCP-47.

But you are actually right anyways, I need to parse the BCP-47 tag first. Probably need to use a proper library.

Reply:
Huh, this is a problem. I wasn't aware of what the POSIX standard was when I wrote the client spec. Might be worth tweaking that part of the spec?

Reply:
Thanks for the link @agnivade makes a lot more sense now, we were looking at BCP-47 and not POSIX. That's a bummer since now that I think about it the POSIX version is probably the only one you'll find used around when it comes to programming and environment variables :confused:.

Reply:
Sure. We would need to update documentation and rename the folders. Want to do that ?

Reply:
@agnivade I'm conflicted on whether or not to do it. Either way, this should be decided now before clients start supporting languages to create minimum trouble. I'll open an issue on the tldr repo so we can discuss this 🙂

Reply:
I'd advocate changing it now, since barely any clients have implemented language support. The longer we leave it, the more complicated it'll become.....

We could even require that clients support both :P

Reply:
Fixed.